synapse/handlers/federation.py

```diff
@@ -267,6 +279,32 @@ async def _maybe_backfill_inner(
             sorted_backfill_points,
         )
 
+        # If we have no backfill points lower than the `current_depth` then
+        # either we can a) bail or b) still attempt to backfill. We opt to try
+        # backfilling anyway just in case we do get relevant events.
+        if not sorted_backfill_points and current_depth != MAX_DEPTH:
+            logger.debug(
+                "_maybe_backfill_inner: all backfill points are *after* current depth. Backfilling anyway."
+            )
+            return await self._maybe_backfill_inner(
+                room_id=room_id,
+                # We use `MAX_DEPTH` so that we find all backfill points next
+                # time (all events are below the `MAX_DEPTH`)
+                current_depth=MAX_DEPTH,
+                limit=limit,
+                # We don't want to another timing observation from this nested
+                # recursive call. The top-most call can record the time overall
+                # otherwise the smaller one will throw off the results.
+                processing_start_time=None,
+            )
+        elif not sorted_backfill_points and current_depth == MAX_DEPTH:
```

Comment:
I think this is equivalent and easier to follow:
```suggestion

       if not sorted_backfill_points:
```

Reply:
Good shout 👍